JavaScript/FightingGame/app.js

```diff
@@ -0,0 +1,400 @@
+class CharacterImage {
+  constructor(characterName) {
+    this.element = document.createElement("img");
+    this.animations = {
+      idle: null,
+      attack1: null,
+      attack2: null,
+      attack3: null,
+      defend: null,
+      defeat: null,
+      damage: null,
+    };
+
+    return this.calcAnimations({ characterName });
+  }
+
+  async calcAnimations({ baseUrl = "./assets/characters", characterName }) {
+    const animationNames = Object.keys(this.animations);
+    const durations = animationNames.map((animationName) => {
+      return this.#calcAnimationDuration({
+        baseUrl,
+        characterName,
+        animationName,
+      });
+    });
+
+    await Promise.all(durations);
+    return this;
+  }
+
+  loadImage(animationName) {
+    const blob = new Blob([this.animations[animationName].arrayBuffer], {
+      type: "image/gif",
+    });
+    const imageUrl = URL.createObjectURL(blob);
+    this.element.src = imageUrl;
+    this.element.alt = animationName;
+  }
+
+  async #calcAnimationDuration({ baseUrl, characterName, animationName }) {
+    return fetch(`${baseUrl}/${characterName}/${animationName}.gif`)
+      .then((res) => res.arrayBuffer())
+      .then((ab) => {
+        return (this.animations[animationName] = {
+          duration: this.#isGifAnimated(new Uint8Array(ab)),
+          arrayBuffer: new Uint8Array(ab),
```

Comment:
please not forgot about try catch

catch (error) {
        console.error("Failed to load animation:", error);
    }